### PR TITLE
Import ordering

### PR DESCRIPTION
**Pull request by apostasie**
This here expands on re-ordering imports for increased code legibility and import assessment, and provides tooling to help developers, and enforce ordering on the CI.

Commit number 1 is just the actual reordering.

Commit number 2 has the tooling and documentation.

In a shell:
- `make lint-imports` will yell at you if ordering is not right
- `make lint-fix-imports` will fix the ordering for you

The current ordering will be:

```
import (
   "gopackages"
   "gopackages"

   "non_containerd_packages"
   "non_containerd_packages"

   "containerd_packages"
   "containerd_packages"

   "nerdctl_packages"
   "nerdctl_packages"
)
```

This order here is configurable, and if we would prefer another way to order imports, I believe we can just do that.

The tool being used is https://github.com/incu6us/goimports-reviser

cc @fahedouch thanks for pointing out ^

### CODE REVIEW

File: docs/contributing.md
Comment (AkihiroSuda) [inline docs/contributing.md:1]: Thanks, but probably this should be a separate PR

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## Before you start actual work
+
+- _look into the issue tracker and PRs_ to see if something similar has been reported or attempted before
+- if _you are fixing a bug, start by opening a bug report_ (if you cannot find one) that explains the problem
+- if _you are developing a new feature, start by opening a feature proposal ticket_ (if you cannot find one)
+that explains what you want to do and why we should do it
+
+Make sure your environment is set-up. Besides go (1.22+) and a working containerd installation, you need
+- golangci-lint
+- goimports-reviser
```
Comment (AkihiroSuda) [inline docs/contributing.md:12]: This paragraph looks slightly misleading, as these tools are only needed for linting

File: docs/testing.md [hunk 2]
```diff
@@ -1,14 +1,5 @@
 # Testing nerdctl
 
-## Lint
```
Comment (AkihiroSuda) [inline docs/testing.md:3]: I guess lint can stay here?

**Comment by apostasie** [comment]
@AkihiroSuda removed `md` changes and will split that out in a separate documentation PR.

**Review by AkihiroSuda — APPROVED** [review]
Thanks
